### PR TITLE
Add approved opportunities to pending page

### PR DESCRIPTION
**Pull request by hamhands**
### What changed
- added view of opportunities approved by omb but not advertised yet
### What's needed
- [x] gate for approved, but not-yet-public contracts
- [x] sort by advertise date
- [x] update tests
- [x] write new test
### Issues
- #195 
### Screencap

screen shot 2015-08-11 at 2 58 37 pm

### CODE REVIEW

**Comment by bsmithgall** [comment]
The border surrounding the table looks a bit weird.

File: purchasing_test/unit/opportunities/test_opportunity_admin.py [hunk 1]
```diff
@@ -423,6 +423,18 @@ def test_pending_opportunity_admin(self):
...
+        '''Test approved opportunities work as expected for city staff
+        '''
+        self.login_user(self.admin)
+        admin_publish = self.client.get('/beacon/admin/opportunities/{}/publish'.format(
+            self.opportunity1.id
+        ))
+        self.assert_flashes('Opportunity successfully published!', 'alert-success')
+        self.assertEquals(admin_publish.status_code, 302)
+        self.assertTrue(Opportunity.query.get(self.opportunity1.id).is_public)
+        self.assertFalse(Opportunity.query.get(self.opportunity1.id).is_advertised)
```
Comment (bsmithgall) [inline purchasing_test/unit/opportunities/test_opportunity_admin.py:436]: You'll want to add the test we talked about -- logged out users shouldn't be able to get to the detail page for this opportunity, but staff should be able to.

```
self.assert200(self.client.get('/beacon/opportunities/{}'.format(self.opportunity1.id))
self.logout_user()
self.assert404(self.client.get('/beacon/opportunities/{}'.format(self.opportunity1.id))
```

or something along those lines

**Comment by hamhands** [comment]
Sorted by advertise date:
screen shot 2015-08-11 at 5 16 34 pm

**Comment by hamhands** [comment]
Ready for review @bsmithgall @shellicious!

**Comment by bsmithgall** [comment]
"Advertised" as a column header feels ambiguous. It should probably reflect the fact that it isn't public/doesn't exist yet.

**Comment by hamhands** [comment]
@bsmithgall so like "Made public" or "Published"?

**Comment by hamhands** [comment]
What about "public on"?
screen shot 2015-08-11 at 5 16 34 pm

**Comment by bsmithgall** [comment]
Looks good to me @hamhands. I think that it would be worthwhile to put a flag on opportunity detail pages that aren't public. Like a little info alert that says something to the effect of "This opportunity is not yet live but will be once it is approved by OMB and/or on its advertised open date."

**Comment by hamhands** [comment]
@bsmithgall yeah that's a really good idea. Merge and then make a new issue
for that?

**Comment by bsmithgall** [comment]
It should be like 5 lines of HTML so I would do it as a part of this.

**Comment by msknee** [comment]
Great!

Can you get rid of the weird border around the table?

**Comment by hamhands** [comment]
Option the first:
screen shot 2015-08-12 at 3 30 48 pm
Option the second:
screen shot 2015-08-12 at 3 30 34 pm

I'm leaning towards the first, but maybe I'm just biased towards the content.

**Comment by msknee** [comment]
I like the 1st :dog: :dog: :dog:

**Comment by bsmithgall** [comment]
+1

**Comment by bsmithgall** [comment]
Is it possible to get the left-align on the alert box right?

**Comment by hamhands** [comment]
Better @shellicious @bsmithgall?
screen shot 2015-08-12 at 3 57 46 pm

**Comment by msknee** [comment]
:thumbsup:

On Wed, Aug 12, 2015 at 3:58 PM, Patrick Hammons notifications@github.com
wrote:

> Better @shellicious https://github.com/shellicious @bsmithgall
> https://github.com/bsmithgall?
> [image: screen shot 2015-08-12 at 3 57 46 pm]
> https://cloud.githubusercontent.com/assets/3640800/9238875/e60a68ec-410a-11e5-9407-90f17acc5aba.png
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codeforamerica/pittsburgh-purchasing-suite/pull/202#issuecomment-130472023
> .

## 

_SHELLY X NI_
Code for America Fellow
509 842 8457 | averyseriousdesigner.com

**Comment by hamhands** [comment]
I regret asking this but...do we need to write tests for all of the messages that come up on the detail page? Like when an opportunity is pending, not advertised, open, closed, etc.?

**Comment by bsmithgall** [comment]
Given the most recent commit, it's probably not a bad idea, but we'll merge it given that the Opportunity model has really high test coverage.

**Pull request merged**